### ISSUE TITLE
[TALENT] Ardent Defender-NO PROC (again)

### ISSUE THREAD

**Issue by wowmania-Kindly** [WoWManiaUK/Redemption#3059]
Hey Again!
So Ardent defender is not functioning well lately. In my Guilds raids we have many times noticed it not procing. This Talent is a barrier against a raidwipe, i hope you can figure out the problem.
Anyway here are some screenshots of one of our Paladin tanks _**immediatly**_ after his Death on Halion 25hc on our sunday raid (17th febuary). I post them only to demonstrate the bug is real.
![wowscrnshot_021719_150624](https://user-images.githubusercontent.com/31795370/52979100-6dd0e200-33d4-11e9-8b37-c40d73e6d096.jpg)
![wowscrnshot_021719_150638](https://user-images.githubusercontent.com/31795370/52979157-aa9cd900-33d4-11e9-9a66-5d5d33ca766a.jpg)
![wowscrnshot_021719_140935](https://user-images.githubusercontent.com/31795370/52979162-af618d00-33d4-11e9-8d25-62d8f39d1de0.jpg)
![wowscrnshot_021719_141928](https://user-images.githubusercontent.com/31795370/52979164-b25c7d80-33d4-11e9-830b-e9fd3deffab7.jpg)

He tells me, the Debuff left by Ardent Proccing, will persist through Death, even while dead, so it should have been present on him on these screenshots. 
He tells me it fails to proc, sometimes from melee hits (physical dmg) and sometimes from breaths (shadow dmg)
The hits from Halion 25hc are very high numbers, but he says he has also experienced the same bug on 5m Dungeons from trash.
He also says on one attempt, Ardent worked once first, then after the debuff had gone ardent failed.
He also tried relogging, spec swapping back and forth, no Luck With either.

In short: Ardent sometimes works as intended, and sometimes doesnt. 
I wish i could present a pattern of when it fails, but we have been unable to find it.
I hope you can figure something out! Thank you for Your work on the server once again.

**Comment by Kitzunu** [comment]
Related https://github.com/WoWManiaUK/Redemption/issues/3030

**Comment by Dranah** [comment]
Will investigate further, but we may be looking for a needle in a haystack. If you can add more information on which 5man dungeon trash it has also occured, that can help find relation.

**Comment by Dranah** [comment]
Ok I think I have it figured... I noticed on your screens the pally have recently used Divine Sacrifice.
I have managed to reproduce the problem with both DS and hand of sacrifice.
It seems transferring damage from your party members with those spells somehow disables the ardent defender proc. It's back to normal if you relog or switch spec.

**How it should work:**
Damage transferred by Divine Sacrifice and Hand of Sacrifice should not interfere with the Ardent Defender proc.
If the transferred damage would kill the paladin, it should be also checked for Ardent being available and proc to prevent the death.
**How it works atm:**
Any damage transferred by DS or HoS bugs the Ardent proc and prevents it from working until the paladin relogs or respec.
Does not matter if the effect is terminated early or runs the full duration.

http://www.wow-mania.com/armory/?spell=64205
http://www.wow-mania.com/armory/?spell=6940
http://www.wow-mania.com/armory/?spell=31852

**Comment by Keader** [comment]
Test now.

**Comment by Dranah** [comment]
Transferred damage no longer disables the proc of Ardent Defender.
If the transferred damage would kill the paladin, it can also be prevented by the ardent proc.

**Comment by wowmania-Kindly** [comment]
Great work!
